update_agol_vehicles_pallet.py

```diff
@@ -0,0 +1,201 @@
+'''
+update_agol_vehicles_pallet.py:
+Automates pulling csvs from an FTP folder and updating a hosted feature
+service with the contents of the latest one.
+'''
+
+import datetime
+import os
+import sys
+
+from pathlib import Path
+from urllib.error import HTTPError
+
+import arcgis
+import arcpy
+import pysftp
+
+from forklift.models import Pallet
+
+import fleetshare_secrets as secrets
+
+
+def get_latest_csv(temp_csv_dir, log, previous_days=-1):
+    '''
+    Returns the Path object and date of the latest 'vehicle_data_*.csv' file in
+    temp_csv_dir. Will fail if limit_three_days is True and the date on the
+    latest csv does not fall within the three preceding days.
+    '''
+
+    #: get list of csvs
+    temp_dir_path = Path(temp_csv_dir)
+    csvs = sorted(temp_dir_path.glob('vehicle_data_*.csv'))
+
+    #: The last of the sorted list of csvs should be the latest
+    try:
+        latest_csv = csvs[-1]
+    except IndexError as e:
+        log.error('Can\'t get last "vehicle_data_*.csv" file- are there any'
+                  ' csv files?')
+        raise e
+
+    #: Pull the date out of vehicle_data_yyyymmdd.csv to check recency
+    date_string = str(latest_csv).rsplit('_')[-1].split('.')[0]
+    try:
+        csv_datetime = datetime.date(int(date_string[:4]),
+                                     int(date_string[4:6]),
+                                     int(date_string[6:]))
+    except ValueError as e:
+        log.error(f'Can\'t parse date from last csv: {latest_csv}')
+        raise e
+
+    #: Only continue if the latest is within specified number of days
+    today = datetime.date.today()
+    previous_dates = [today - datetime.timedelta(days=i)
+                      for i in range(previous_days + 1)]
+    if previous_days > 0 and csv_datetime not in previous_dates:
+        message = (f'Latest csv "{latest_csv}" not within {previous_days} days'
+                   f' of today ({today})')
+        log.info(message)
+        sys.exit(message)
```

Comment:
**suggestion**: I think I would throw here. sys.exit may try to kill the forklift process.